### PR TITLE
Api

### PR DESCRIPTION
**Pull request by atareshawty**
Added an api layer. Pages handler acts as the C in a MVC app and requests all user info from the api through http requests. Client side scripts were moved off the user templates into separate files to clean that up. User can delete him/herself after slight confirmation process. This 'rewrite' makes a testable api, so more tests are coming

### CODE REVIEW

File: db/users.js [hunk 1]
```diff
@@ -217,11 +275,10 @@ function getCategories(id, done) {
  * Callback function {@done} expects err and boolean
  * @param username
  * @param callback function
- */
+ */
```
Comment (tarebyte) [inline db/users.js]: whitespace

File: db/users.js [hunk 2]
```diff
@@ -217,11 +275,10 @@ function getCategories(id, done) {
...
- */
+ */ 
 function usernameExists(username, done) {
-  console.log('Username exists');
-  var client = createDBClient()
-  var query;
+  var client = createDBClient(), query;
+  
   client.on('error', function(err) {
     console.log('Error from username exists: ' + err.message);
```
Comment (tarebyte) [inline db/users.js]: If you're removing all of the logs you may want to do it everywhere

Comment (tarebyte) [inline db/users.js]: If you want to remove the logs everywhere you may want to remove this one.

File: db/users.js [hunk 3]
```diff
@@ -185,6 +228,21 @@ exports.insertNewCategory = function(id, category, done) {
   });
 };
 
+exports.deleteUser = function(id, done) {
+  var client = createDBClient();
```
Comment (tarebyte) [inline db/users.js]: do you have to do this for every call? Seems kind of annoying

File: handlers/APIHandler.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
+var db = require('../db/users.js');
+var User = require('../public/javascripts/userModel.js');
+var Transaction = require('../public/javascripts/transactionModel.js');
+var brcypt = require('bcrypt-nodejs');
+
+function APIHandler() {
+
```
Comment (tarebyte) [inline handlers/APIHandler.js]: I would read this over: http://racksburg.com/choosing-an-http-status-code/ to understand the status codes that you are sending

File: handlers/APIHandler.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+		db.insertUsernameAndPassword(username, password, function(err, inDB) {
+			if (inDB) {
+				res.status(403).send();
+			} else {
+				res.status(200).send();
+			}
+		})
+	};
+	
+	this.getUserModel = function(req, res) {
```
Comment (tarebyte) [inline handlers/APIHandler.js]: `getUser` instead of `getUserModel`

File: db/users.js (hunk 3 above)
Comment (atareshawty) [inline db/users.js]: Yeah, I imagine it can be replaced with one global client. I'll look into it

File: handlers/APIHandler.js (hunk 4 above)
Comment (atareshawty) [inline handlers/APIHandler.js]: I read through the article and changed how I was sending some responses. Thanks @tarebyte!

**Pull request closed without merging**